src/hotspot/share/gc/shared/cardTable.hpp

```diff
@@ -182,6 +183,10 @@ class CardTable: public CHeapObj<mtGC> {
     return _card_shift;
   }
 
+  static uint card_shift_in_words() {
```

Comment:
Seems that this new API is used only for Serial and G1. I wonder if they can be changed to sth like to what Parallel does to drop the dependency on this method.

Reply:
Parallel doesn't need the method `card_shift_in_words` because the parameter type of its method `ObjectStartArray::addr_for_entry` is different from the `SerialBlockOffsetSharedArray::address_for_index` and `G1BlockOffsetTable::address_for_index`.

If you don't want to add this API, I suggest using `CardTable::card_shift() - LogHeapWordSize` to replace `CardTable::card_shift_in_words` in the code. What do you think?

Reply:
> because the parameter type of its method ObjectStartArray::addr_for_entry is different from the SerialBlockOffsetSharedArray::address_for_index and G1BlockOffsetTable::address_for_index.

Could the signatures be unified? I was thinking of a followup PR to address that; this patch looks fine.

Reply:
> Could the signatures be unified? I was thinking of a followup PR to address that; 

Maybe. Have you started the work locally?

Reply:
No; feel free to create a ticket on that.

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8328508 to follow up.